addons/point_of_sale/models/pos_order.py

```diff
@@ -539,6 +539,9 @@ def _default_pricelist(self):
         domain="[('state', '=', 'opened')]", states={'draft': [('readonly', False)]},
         readonly=True, default=_default_session)
     config_id = fields.Many2one('pos.config', related='session_id.config_id', string="Point of Sale", readonly=False)
+    currency_id = fields.Many2one('res.currency', related='config_id.currency_id', string="Currency")
```

Comment:
related='sale_journal.currency_id', it is the same for now, but it is more direct and consistent